extensions/git/src/repository.ts

```diff
@@ -1363,6 +1363,34 @@ export class Repository implements Disposable {
 		await this.run(Operation.RenameBranch, () => this.repository.renameBranch(name));
 	}
 
+	@throttle
+	async fastForwardBranch(name: string): Promise<void> {
+		const config = workspace.getConfiguration('git', Uri.file(this.root));
+		const fetchBeforeCheckout = config.get<boolean>('fetchBeforeCheckout', false) === true;
+
+		if (!fetchBeforeCheckout) {
+			return;
+		}
+
+		// Get branch details
+		const branch = await this.getBranch(name);
```

Comment:
We actually have this information up in `CheckoutItem.run()`, so why not passing along a `ref` here? Other methods from this class which follow that pattern are `push()` and `sync()`.

Reply:
As discussed, `CheckItem.run()` uses a `Ref` and not a `Branch` object.